### ISSUE TITLE
[AVM Module Issue]: management-group has no option for diagnostic logs

### ISSUE THREAD

**Issue by awood86** [Azure/bicep-registry-modules#3362]
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Feature Request

### Module Name

avm/res/management/management-group

### (Optional) Module Version

0.1.2

### Description

Would you be able to add Diagnostics onto this module. A bit like below:

targetScope = 'managementGroup'

param parDiagnosticSettingsName string = 'toLa'

param parLogAnalyticsWorkspaceResourceId string

resource resMgDiagSet 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = {
  name: parDiagnosticSettingsName
  properties: {
    workspaceId: parLogAnalyticsWorkspaceResourceId
    logs: [
      {
        category: 'Administrative'
        enabled: true
      }
      {
        category: 'Policy'
        enabled: true
      }
    ]
  }
}

**Comment by fblix** [comment]
Since I am currently quite busy, I would appreciate any help on this topic!

**Comment by AlexanderSehr** [comment]
I'll be damned. It seems like there is an [API](https://learn.microsoft.com/en-us/rest/api/monitor/management-group-diagnostic-settings/create-or-update?view=rest-monitor-2020-01-01-preview&tabs=HTTP) even though I cannot seem to find it in the Portal. In any case, that means nothing should speak against implementing it. The only challenge I see is that it's a bit difficult to identify which metrics/logs can be configurated.

**Comment by awood86** [comment]
> I'll be damned. It seems like there is an [API](https://learn.microsoft.com/en-us/rest/api/monitor/management-group-diagnostic-settings/create-or-update?view=rest-monitor-2020-01-01-preview&tabs=HTTP) even though I cannot seem to find it in the Portal. In any case, that means nothing should speak against implementing it. The only challenge I see is that it's a bit difficult to identify which metrics/logs can be configurated.

Yeah it doesn't show in the portal for some reason. I picked up the diagnostic logs for the Management Groups from the ALZ Bicep modules as they provide this functionality

Just one to add to the module notes:

Notes
Considerations
This template is using a Tenant level deployment, meaning the user/principal deploying it needs to have the [proper access](https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/deploy-to-tenant#required-access)

If owner access is excessive, the following rights roles will grant enough rights:

Automation Job Operator at tenant level (scope '/')
Management Group Contributor at the top management group that needs to be managed
Consider using the following script:

$PrincipalID = "<The object ID of the identity here>"
$TopMGID = "<The group ID of the management group here>"
New-AzRoleAssignment -ObjectId $PrincipalID -Scope "/" -RoleDefinitionName "Automation Job Operator"
New-AzRoleAssignment -ObjectId $PrincipalID -Scope "/providers/Microsoft.Management/managementGroups/$TopMGID" -RoleDefinitionName "Management Group Contributor"

The user/principal will also need something like Monitoring Contributor to the Management Group and will also need something like Log Analytics Contributor to the workspace (If they are deploying to the Log Analytics workspace)

**Comment by AlexanderSehr** [comment]
Hey @fblix,
Please triage the issue when you get the chance 🙂

**Comment by jikuja** [comment]
I just tested this with `PUT https://management.azure.com/providers/microsoft.management/managementGroups/testChildMG7/providers/microsoft.insights/diagnosticSettings/setting1?api-version=2020-01-01-preview`  with payload

```
{
  "properties": {
    "workspaceId": "/subscriptions/xxxxxxx-7abd-4d40-a1ad-feec498aac80/resourcegroups/xxxxxx-infra/providers/microsoft.operationalinsights/workspaces/xxxxx-law",
    "logs": [
      {
        "enabled": true,
        "category": "Administrative"
      },
      {
        "enabled": true,
        "category": "Policy"
      }
    ]
  },
  "location": "westeurope"
}
```

API request returned 200 OK and resulting payload was correct. 

-----

Log analytics workspace I used have not yet received any logs from management. Therefore it might be a good idea to check from PG or landing zone developers if this works or not. 

cc: @fblix @AlexanderSehr

**Comment by jikuja** [comment]
Looks like there was huge delay before log entries were copied into LAW after diagnostics setting deployment. Anyway knowing the categories would be nice.
